src/rest/APIRequest.js

```diff
@@ -46,12 +47,15 @@ class APIRequest {
       headers['Content-Type'] = 'application/json';
     }
 
+    const controller = new AbortController();
+    const timeout = setTimeout(() => controller.abort(), this.client.options.restRequestTimeout);
```

Comment:
```suggestion
    const timeout = this.client.setTimeout(() => controller.abort(), this.client.options.restRequestTimeout);
```

`this.client` is available at:

https://github.com/discordjs/discord.js/blob/5c0d26bd12ad338280de7ea9626ba52625e0df5c/src/rest/APIRequest.js#L14

And this change allows `Client#destroy()` to clear all the timeouts.

Reply:
Thanks for noticing that, I actually didn't thought about making it as an timeout in client instead. Will be patching things later.

Reply:
PR Updated to use d.js timeout methods instead 👍